File: app/src/main/java/de/christinecoenen/code/zapp/utils/system/MultiWindowHelper.java
package de.christinecoenen.code.zapp.utils.system;


import android.annotation.TargetApi;
import android.app.Activity;
import android.os.Build;

public class MultiWindowHelper {

	/**
	 * This function can be used with any API level and will return
	 * false if the multi window feature is not supported.
	 * @param activity to get access to multi window api
	 * @return true if activity is currently displayed in multi window mode
     */
	@TargetApi(24)
	public static boolean isInsideMultiWindow(Activity activity) {
		return Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && activity.isInMultiWindowMode();
	}

}

File: app/src/main/java/de/christinecoenen/code/zapp/utils/system/MultiWindowHelper.kt
package de.christinecoenen.code.zapp.utils.system

import android.app.Activity
import android.app.PictureInPictureParams
import androidx.appcompat.app.AppCompatActivity

object MultiWindowHelper {
	/**
	 * This function can be used with any API level and will return
	 * true if the activity is currently in pip mode.
	 *
	 * @param activity to get access to pip api
	 * @return true if activity is currently displayed in picture in picture mode
	 */
	@JvmStatic
	fun isInPictureInPictureMode(activity: Activity): Boolean {
		return activity.isInPictureInPictureMode
	}

	@JvmStatic
	fun enterPictureInPictureMode(activity: AppCompatActivity, params: PictureInPictureParams? = null) {
		val nullCheckedParams = params ?: PictureInPictureParams.Builder().build()
		activity.enterPictureInPictureMode(nullCheckedParams)
	}
}
